### DISCUSSION TITLE
Distributed Deployment with Fleet on Client Network

### DISCUSSION THREAD

**Discussion by EricHStevens** [Security-Onion-Solutions/securityonion#5365] [Unsupported Versions] [answered]
We have been trying to setup a distributed deployment of Security Onion where we have the Fleet Node on the client network. The Fleet Node collects the logs from he client machines and then forwards the logs to the Master Node on our provider network. We are able to ping the Fleet from the provider network and the Fleet node can ping the Master node on the provider network. The problem is, in this configuration the Fleet cannot see the machines on the client LAN.
We are guessing we have something configured wrong in the Fleet, just haven't figured out where to look yet. Any one have ideas on configuration settings we could check? Has anyone deployed this way before?

Currently, we are getting around the problem by having all the OsQuery Agents enrolled with Fleet on the provider network side. This works, but to make the design more resilient we would prefer to move the Fleet onto the client side so that in the case of networks outages the client machines can continue to forward logs to the Fleet on their network.

**Comment by defensivedepth** [comment]
A couple questions:

- Is there any kind of NAT between the Fleet node and the Manager?

- Can you post a sanitized output of the following? (Run from the Manager)

`sudo salt-call pillar.get global`

**Reply by EricHStevens** [reply]
Effectively, we did a fresh install of Security Onion 2.3.80 for both the manager, fleet, made sure to do an so-allow, and setup a firewall where the original IP was visible on both sides of the tunnel and things worked.

I believe we can consider this question answered and closed.

**Comment by EricHStevens** [comment, marked as answer]
Effectively, we did a fresh install of Security Onion 2.3.80 for both the manager, fleet, made sure to do an so-allow, and setup a firewall where the original IP was visible on both sides of the tunnel and things worked.

I believe we can consider this question answered and closed.
